### ISSUE TITLE
Parser \n\n behaviour as demoed in RealCsvParserTest

### ISSUE THREAD

**Issue by boardtc** [alisonthemonster/GratitudeJournal#168]
Looking at RealCsvParserTest the parser code deals with new lines being imported in entryContent.

However, here is an example of it not working:

csv file:
2021-01-25,I am grateful I got new kitchen gas & firewood which made me feel responsible\n\nI am grateful I had tuna melt in applegreen which made me feel it was something different\n\nI am grateful I figured out regex replace to change to lowercase which made me feel i'm learning\n\nI am grateful I signed up for Real Python for 1 month which made me feel i'm taking a positive step\n\nI am grateful I learned what a lamba function was in Python which made me feel why did I wait so long

When I import this it does not parse the \n's correctly:

![image](https://user-images.githubusercontent.com/214808/107398944-131c7d80-6af8-11eb-9b36-e12ec6d5a0aa.png)

**Comment by alisonthemonster** [comment]
Hiya Tom!

So looking at this, it looks like the tests are actually showing what is happening in your screenshot, but that's still not great behavior.

From looking at how the tests behave, if you have a single new line character that gets parsed as a new entry. But multiple new line characters in a row gets preserved, see line [59 in RealCsvParserTest](https://github.com/alisonthemonster/Presently/blob/develop/app/src/test/java/journal/gratitude/com/gratitudejournal/util/backups/RealCsvParserTest.kt#L56).

So that's why your new line characters have made it all the way to the UI. The CSV creating library I'm using is doing some escaping (From [StringEscapeUtils](https://github.com/apache/commons-lang/blob/master/src/main/java/org/apache/commons/lang3/StringEscapeUtils.java#L760))
> If the value contains a comma, newline or double quote, then the String value is returned enclosed in double quotes.

There might be a way I can get the TextViews to render this new line char as an actual new line but I'm not sure. 

I've been playing around with unit tests and manual importing of all sorts of escaped characters and I'm not able to have success yet. I know its possible because if I write an entry with new lines in it using the device keyboard enter button, export them to CSV, then import them again the new lines are preserved! I need to dig more into what the Apache CSV library is doing to preserve these new lines. I'll keep trying things and let you know!

**Comment by boardtc** [comment]
Thanks for the detailed explanation with links Alison, very informative!
I updated the issue title since it's expected behaviour.
I edited the above entry in the app to add the new lines where the \n\n are and I exported to see the output as you describe. 
_Hopefully_ I can prepare my csv to match this double quote wrapped behaviour.
Personally I agree that the expected behaviour is a little lame but I'd say it's a rainy day backlog item since it's not affecting imports/exports generated from the app!
Feel free to close.

Update: I actually had a bug in my csv writing code which escaped the \n so it wrote out literally instead of doing a new line! The [python csv](https://docs.python.org/3.8/library/csv.html) handles newlines/quotes the same as apache commons. An import into Presently is now looking perfect!

Good learning, thanks.
